lib/components/table.vue

```diff
@@ -524,14 +533,25 @@
 
                 if (data.then && typeof data.then === 'function') {
                     // Provider returned Promise
-                    data.then((items) => {
+                    data.then(items => {
                         this._providerSetLocal(items);
                     });
                 } else {
                     // Provider returned Array data
                     this._providerSetLocal(data);
                 }
+            },
+            hasFormatter(field) {
+                return field.formatter && ((typeof (field.formatter) === 'function') || (typeof (field.formatter) === 'string'));
+            },
+            callFormatter(item, key, field) {
+                if (field.formatter && (typeof (field.formatter) === 'function'))
+                    return field.formatter(item[key]);
+
+                if (field.formatter && (typeof (this.$parent[field.formatter]) === 'function'))
+                    return this.$parent[field.formatter](item[key]);
```

Comment:
Then we will have two ways of handling table data: slots and formatter callback.
Callbacks will give more reusability, as we can move formatter functions out to standalne module and use it as mixin at components and even as global mixin to app's Vueinstance, 
while slots will allow to insert another components into table cell and also format field value and create virtual columns in case we don't need to reuse it at whole app.